Bot finished

KarataevDmitry opened the pull request:
closes #4

EvilLord666 committed:
Update README.md

KarataevDmitry committed:
Code finished. Tests in progress

KarataevDmitry committed:
Test passed

KarataevDmitry committed:
Merge branch 'Wissance:master' into master

KarataevDmitry committed:
set template ext as txt

EvilLord666 committed:
Merge branch 'develop' into master

EvilLord666 commented:
Conflicts were resolved

EvilLord666 commented on Wissance.MossbauerLab.Watcher/Wissance.MossbauerLab.Watcher.Services/Notification/NotificationMessageFormatter.cs:
```diff
@@ -0,0 +1,37 @@
+    {
+        private const string AutosavedSpectraPlaceholder = "{savedSpectra}";
+        // <!--<li>Спектр {msSpName} по каналу {msChNumber} сохранен {msSaveDate}</li>-->
+        private const string SavedSpectrumDescriptionTemplate = "<li>Спектр {0} по каналу {1} сохранен {2}</li>";
```
By rules of Wissance projects const should be placed after methods (below)

EvilLord666 commented on Wissance.MossbauerLab.Watcher/Wissance.MossbauerLab.Watcher.Services/Notification/TelegramNotifier.cs:
```diff
@@ -1,11 +1,49 @@
-    public class TelegramNotifier
+    public class TelegramNotifier : ISpectrumReadyNotifier
     {
+        private readonly ApplicationConfig _config;
```
Field && consts should be placed below methods

EvilLord666 commented on Wissance.MossbauerLab.Watcher/Wissance.MossbauerLab.Watcher.Services/Notification/TelegramNotifier.cs:
```diff
@@ -1,11 +1,49 @@
+        {
+            ITelegramBotClient client = new TelegramBotClient("6253527316:AAHrOysU6QKubqCg51aHuVCNBf3PtD3nGBU");
+           
+            var targetGroupName = _config.NotificationSettings.TelegramSettings.NotificationGroupName;
```
We don't use var's either because we always know our types and this helps to make code more readable

KarataevDmitry replied:
Ok, add editor config with codestyle preferences

KarataevDmitry replied:
You can use type hints. In my IDE code looks like this:
![image](https://user-images.githubusercontent.com/2220280/218772417-99db7dc0-5663-4d26-a0f3-50fef44df3d5.png)

EvilLord666 commented on Wissance.MossbauerLab.Watcher/Wissance.MossbauerLab.Watcher.Services/Notification/TelegramNotifier.cs:
```diff
@@ -1,11 +1,49 @@
+            {
+                await client.SendTextMessageAsync(targetChatId, msg.Text);
+            }
+            catch (Exception)
```
When we got an exception we should `LogError` it

EvilLord666 commented on Wissance.MossbauerLab.Watcher/Wissance.MossbauerLab.Watcher.Tests/TelegramNotificationsTests.cs:
```diff
@@ -0,0 +1,45 @@
+        [TestMethod]
+        public async Task SendNotificationTest()
+        {
+            var config = new ApplicationConfig();
```
Replace var with an actual type in this test method too

KarataevDmitry committed:
remarks done
Also I fix mistyping in SpectrumEntity.ctor
Property assigned to parameter. It's strange. We assign parameter to property in ctor usually
TelegramNotifier added to DI
